pkgs/build-support/go/module.nix

```
@@ -2,149 +2,152 @@
-# Whether to delete the vendor folder supplied with the source.
+  # Whether to delete the vendor folder supplied with the source.
 , deleteVendor ? false
-# Whether to fetch (go mod download) and proxy the vendor directory.
-# This is useful if your code depends on c code and go mod tidy does not
-# include the needed sources to build or if any dependency has case-insensitive
-# conflicts which will produce platform dependant `vendorHash` checksums.
+  # Whether to fetch (go mod download) and proxy the vendor directory.
+  # This is useful if your code depends on c code and go mod tidy does not
+  # include the needed sources to build or if any dependency has case-insensitive
+  # conflicts which will produce platform dependant `vendorHash` checksums.
 , proxyVendor ? false
 
-# We want parallel builds by default
+  # We want parallel builds by default
 , enableParallelBuilding ? true
 
-# Do not enable this without good reason
-# IE: programs coupled with the compiler
+  # Do not enable this without good reason
+  # IE: programs coupled with the compiler
 , allowGoReference ? false
 
 , CGO_ENABLED ? go.CGO_ENABLED
 
-, meta ? {}
+, meta ? { }
 
-# Not needed with buildGoModule
+  # Not needed with buildGoModule
 , goPackagePath ? ""
 
-# needed for buildFlags{,Array} warning
+  # needed for buildFlags{,Array} warning
 , buildFlags ? ""
 , buildFlagsArray ? ""
 
-, ... }@args':
+, ...
+}@args':
 
 assert goPackagePath != "" -> throw "`goPackagePath` is not needed with `buildGoModule`";
 assert (args' ? vendorHash && args' ? vendorSha256) -> throw "both `vendorHash` and `vendorSha256` set. only one can be set.";
 
 let
   args = removeAttrs args' [ "overrideModAttrs" "vendorSha256" "vendorHash" ];
 
-  go-modules = if (vendorHash != null) then stdenv.mkDerivation (let modArgs = {
-
-    name = "${name}-go-modules";
-
-    nativeBuildInputs = (args.nativeBuildInputs or []) ++ [ go git cacert ];
-
-    inherit (args) src;
-    inherit (go) GOOS GOARCH;
-
-    # The following inheritence behavior is not trivial to expect, and some may
-    # argue it's not ideal. Changing it may break vendor hashes in Nixpkgs and
-    # out in the wild. In anycase, it's documented in:
-    # doc/languages-frameworks/go.section.md
-    prePatch = args.prePatch or "";
-    patches = args.patches or [];
-    patchFlags = args.patchFlags or [];
-    postPatch = args.postPatch or "";
-    preBuild = args.preBuild or "";
-    postBuild = args.modPostBuild or "";
-    sourceRoot = args.sourceRoot or "";
-
-    GO111MODULE = "on";
-
-    impureEnvVars = lib.fetchers.proxyImpureEnvVars ++ [
-      "GIT_PROXY_COMMAND" "SOCKS_SERVER" "GOPROXY"
-    ];
-
-    configurePhase = args.modConfigurePhase or ''
-      runHook preConfigure
-      export GOCACHE=$TMPDIR/go-cache
-      export GOPATH="$TMPDIR/go"
-      cd "${modRoot}"
-      runHook postConfigure
-    '';
+  go-modules =
+    if (isNull vendorHash) then "" else
```

Sorry, in my previous comment I meant revert this:
```suggestion
  go-modules = if (isNull vendorHash) then "" else
```